sql/src/main/java/io/crate/action/sql/SQLOperations.java

```diff
@@ -143,8 +143,8 @@ public Session createSession(@Nullable String defaultSchema, Set<Option> options
         private Statement statement;
         private String query;
 
-        private Map<String, Portal> portals = new HashMap<>();
-        private String lastPortalName;
+        private final Map<String, Portal> portals = new HashMap<>();
+        private final Set<Portal> pendingExecutions = Collections.newSetFromMap(new IdentityHashMap<Portal, Boolean>());
```

Comment:
Currently there should be at most 1 entry so it doesn't really matter - I can also change it to a HashSet for now if you want